addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -707,6 +732,35 @@ import { sprintf } from "@web/core/utils/strings";
                 inputEl.disabled = !haveToBeVisible;
             }
         },
+        /**
+         * Creates a block containing the file name and a cross to delete it.
+         *
+         * @private
+         * @param {Object} fileDetails the details of the file being uploaded
+         * @param {Element} filesZoneEl the zone where the file blocks are
+         *      displayed
+         */
+        _createFileBlock(fileDetails, filesZoneEl) {
+            const fileBlockEl = qweb.render('website.file_block', {fileName: fileDetails.name});
+            filesZoneEl.insertAdjacentHTML('beforeend', fileBlockEl);
+            filesZoneEl.lastElementChild.fileDetails = fileDetails;
+        },
+        /**
+         * Creates the file upload button (= a button to replace the file input,
+         * in order to modify its text content more easily).
+         *
+         * @private
+         * @param {Element} inputEl the file input
+         */
+        _createAddFilesButton(inputEl) {
+            const addFilesButtonEl = document.createElement('INPUT');
+            addFilesButtonEl.classList.add('o_add_files_button', 'form-control');
+            addFilesButtonEl.type = 'button';
+            addFilesButtonEl.title = inputEl.title;
+            addFilesButtonEl.value = inputEl.getAttribute('multiple') != null ? _t("Add Files") : _t("Replace File");
```

Comment:
Shouldn't we disabled those buttons if the the number of files is already equal to the max ?

Reply:
I can ask PDE but I don't think it is needed because there are cases where it will not make sense :thinking:  
For example:
- If only one file is allowed, then uploading a file is already the max and the button is supposed to replace the file. But if we disable the input, then we can't replace it -> we will have to delete the file first before uploading the new one, which is not convenient
- The `multiple` attribute allows to select as many files as we want with no limit. So, if I set a limit to 3 files and upload 4, the input will be disabled but the number of files is not correct. I find it confusing if there is no warning
- I think it is better to let the user choose freely as many files as he wants and then delete if he uploaded too much (the limits are supposed to be described by the "admin" and there is the tooltip anyway) :)